packages/reakit/src/Popover/__examples__/HoverPopover/index.tsx

```diff
@@ -0,0 +1,72 @@
+import * as React from "react";
+import { Button } from "reakit/Button";
+import {
+  usePopoverState,
+  Popover,
+  PopoverDisclosure,
+  PopoverArrow,
+  PopoverInitialState,
+} from "reakit/Popover";
+import { Separator } from "reakit/Separator";
+
+function useHoverPopoverState(initialState?: PopoverInitialState) {
+  let timeout: number;
```

Comment:
Have you checked if it's working? I'm pretty sure it should be a ref to work across renders.

Reply:
yes, it is 🙄

Reply:
Hmm, yeah! It's working in this case because there's no intermediate render that affects `show` in the interval of <300 ms between the `hideWithDelay` call and the `show` call. But this may not be true in the context of the users' apps.

Reply:
I've changed for ref usages instead, I added a timeout for the `show` fn too